PublishParasoftTestResults/ParaReportPublishService.ts

```diff
@@ -356,6 +357,48 @@ export class ParaReportPublishService {
         }
     }
 
+    processParasoftSarifReport = (report: string): string => {
+        let contentString = fs.readFileSync(report, 'utf8');
+        let contentJson = JSON.parse(contentString);
+        let isContentChanged = false;
+        if (contentJson.runs) {
+            contentJson.runs.forEach((run: any) => {
+                if (run.results) {
+                    run.results.forEach((result: any) => {
+                        if (result.locations) {
+                            result.locations.forEach((location: any) => {
+                                if (location.physicalLocation && location.physicalLocation.artifactLocation 
+                                        && location.physicalLocation.artifactLocation.uri) {
+                                    let uri: string = location.physicalLocation.artifactLocation.uri;
```

Comment:
Is it better to extract this block of code into a method as "getArtifactUri" and assign it to "location.physicalLocation.artifactLocation.uri" to have better readability of the code?